timemachine/fe/mcgregor.py

```diff
@@ -367,43 +421,24 @@ def recursion(
     if num_edges < threshold:
         return
 
-    if connected_core:
-        # process g1 using atom_map_1_to_2_information
-        g1_node_states = [NODE_STATE_VISITED_DEMAPPED] * g1.n_vertices
-        g1_source = None
-        g1_mapped_count = 0
-        for a1, a2 in enumerate(atom_map_1_to_2):
-            if a1 < layer:
-                # visited nodes
-                if a2 != UNMAPPED:
-                    g1_node_states[a1] = NODE_STATE_VISITED_MAPPED
-                    g1_source = a1
-                    g1_mapped_count += 1
-            else:
-                g1_node_states[a1] = NODE_STATE_UNVISITED
-
-        if g1_source and g1.mapping_is_disconnected(g1_source, g1_node_states, g1_mapped_count):
-            return
+    if max_connected_components is not None or min_connected_component_size > 1:
+        g1_mapped_nodes = {a1 for a1, a2 in enumerate(atom_map_1_to_2[:layer]) if a2 != UNMAPPED}
```

Comment:
Nice improvement to the code, but with it being so much more concise it would be useful to have a brief comment about the intention

Reply:
Added comments in https://github.com/proteneer/timemachine/pull/1349/commits/b27be3af5e0b6c90391ccfcee77dcee4bfb0fad4; hopefully these make the logic clearer